### ISSUE TITLE
Disable static analyzer by default

### ISSUE THREAD

**Issue by cusma** [algorandfoundation/algokit-python-template#73]
## Problem

The `.algokit/static_analysis/snapshot` JSONs keep growing (quickly) with the `#` of `pre-commit` runs.

I've noticed it just when GB rejected a commit due to file oversize (118 MB 😱).

## Solution

Make static analysis optional and disable them by default.

## Dependencies

None.

**Comment by aorumbayev** [comment]
@cusma can you provide an example or expand on what exactly is growing quickly? 
It should match the # of artifacts in the artifacts folder. So if you add N contracts - upon running tealer it will save N snapshots - which are then used in ci to compare if the pushed contract has any new results from tealer output compared against snapshots. 

I have just instantiated a template against latest python example, generated 100 hello world artifacts, and then run the analyze task which resulted in a snapshots folder of size only ~2mb. Which will remain static unless new contracts are added or new content is added into the template. 

Are you able to provide a small reproducible example if you are saying that multiple re  runs of the tealer task against unchanged artifacts increases size of snapshots. Alternatively if you have a lot of contracts, it may also depend on the complexity of the contract itself, algokit-cli only forwards the inputs into tealer but if tealer produces an abnormally large snapshot file then might be an even better case for having a reproducible example so we can report on the tealer repository.

**Comment by cusma** [comment]
So, I can be wrong here. Not sure if I'm able to set up a reproducible example quickly, but... this is my condition: I have just 1 contract. The resulting `.approval.json` snapshot is 118 MB size. The smart contract is not so complex. So even if it's static I think 118 MB is pretty heavy, considered that AlgoKit project are usually committed to GH.

**Comment by aorumbayev** [comment]
@cusma this sounds like a bug on tealer side then, any chance this contract can be shared ? I can re test against tealer and open an issue on tealer repository with a reproducible snippet.

**Comment by aorumbayev** [comment]
@cusma just found this on tealer repo, looks like its the related root cause for why your approval snapshot is so large -> https://github.com/crytic/tealer/issues/101

**Comment by aorumbayev** [comment]
@cusma above PR disables this as per this issue but will keep it open until there are fixes in tealer. At that point worth re enabling the behaviour as other than the existing bug, its a useful feature to have out of the box

**Comment by cusma** [comment]
> @cusma just found this on tealer repo, looks like its the related root cause for why your approval snapshot is so large -> https://github.com/crytic/tealer/issues/101

Yes, that my be the cause. My contract is not complex but involved quite heavy loops. Maybe TEALer files/paths proliferate with each loop execution.
